modules/Hoot/ui/sidebar/advancedOpts.js

```diff
@@ -848,34 +848,30 @@ export default class AdvancedOpts {
         }
 
         function checkType( member ) {
+            let defaultVal;
             if ( member.input === 'checkbox' ) {
-                updateOpts.push(
-                    {
-                        input: member.input,
-                        default: d3.select(`#${member.id}`).select('input').property('checked'),
-                        id: member.id,
-                        description: member.description,
-                        label: member.label,
-                        type: member.type,
-                        option: member.option
-
-                    }
-                );
+                defaultVal = d3.select(`#${member.id}`).select('input').property('checked');
```

Comment:
if the only thing that is different here is 'checked' versus 'value' could you just do a `defaultVal  = ${all that d3 stuff}.property(member.input === checkbox ? 'checked' : 'value')`

Reply:
even better just assign the value in the dictionary `selectedOpt` rather than creating the variable for it

Reply:
great idea